### ISSUE TITLE
PC_FilterBetween gives wrong output

### ISSUE THREAD

**Issue by tomvantilburg** [pgpointcloud/pointcloud#25]
The PC_FilterBetween function appears to do a multiplication by 10 of my z-values, although I've also seen by 100 or more random values. Here's an example to reproduce and the description of my z-dimension from the schema below.

Example:
```
WITH points AS (
SELECT PC_MakePoint(1, ARRAY[z,foo1, foo2, foo3,foo4, foo5, foo6,foo7, foo8, foo9,x,y,foo10,foo11]) pt --sorry for the foos, its due to my schema
FROM (
  SELECT 
  1.0*random() AS z,
  0 as foo1,
  1 as foo2, 
  1 as foo3,
  0 as foo4,
  0 as foo5, 
  0 as foo6,
  0 as foo7,
  0 as foo8, 
  0 as foo9, 
  -127+a/100.0 AS x, 
    45+a/100.0 AS y,
  0 as foo10,
  0 as foo11 
  FROM generate_series(1,100) AS a
) AS values
)

SELECT 
PC_PatchAvg(PC_Patch(pt), 'z'), 
PC_PatchAvg(PC_FilterBetween(PC_Patch(pt), 'z',0, 3),'z') 
FROM points
```

```
<pc:dimension>
  <pc:position>1</pc:position>
  <pc:size>4</pc:size>
  <pc:description>Z coordinate as a long integer. You must use the scale and offset information of the header to determine the double value.</pc:description>
  <pc:name>Z</pc:name>
  <pc:interpretation>int32_t</pc:interpretation>
  <pc:scale>0.01000000000000</pc:scale>
  <pc:active>true</pc:active>
  <pc:uuid>e74b5e41-95e6-4cf2-86ad-e3f5a996da5d</pc:uuid>
  <pc:parent_uuid>00000000-0000-0000-0000-000000000000</pc:parent_uuid>
 </pc:dimension>
```

**Comment by strk** [comment]
your Z values are random numbers from 0 to 100, as per your request: (random()*1.0)/0.01 [scale]

**Comment by strk** [comment]
Actually, the scale should not affect operations (other than overflow). Reopening for closer verification.

**Comment by strk** [comment]
@tomvantilburg you should show the whole schema for debugging this. May be relevant.
I cannot reproduce myself with a schema composed by X,Y,Z all having the same size/intepretation/scale.

**Comment by strk** [comment]
I believe you might have seen the effects of bugs #78 and #79, now fixed in the repository (1.0.1+).
Please reopen if you can still reproduce after an upgrade to that version.
